cf/field.py

```diff
@@ -2374,21 +2378,92 @@ def cell_area(
+                of a field construct.
+
+                .. versionadded:: 3.17.0
+
             insert: deprecated at version 3.0.0
 
             force: deprecated at version 3.0.0
 
         :Returns:
 
-            `Field`
-                A field construct containing the horizontal cell
-                areas.
+            `Field`, `CellMeasure`, or `dict`
+                A field construct, or cell measure construct if
+                *return_cell_measure* is True, containing the
+                horizontal cell areas. If *methods* is True then a
+                dictionary describing the method used to create the
+                cell areas is returned.
 
         **Examples**
 
+        >>> f = cf.example_field(0)
+        >>> print(f)
+        Field: specific_humidity (ncvar%q)
+        ----------------------------------
+        Data            : specific_humidity(latitude(5), longitude(8)) 1
+        Cell methods    : area: mean
+        Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
+                        : longitude(8) = [22.5, ..., 337.5] degrees_east
+                        : time(1) = [2019-01-01 00:00:00]
+
         >>> a = f.cell_area()
+        >>> a
+        <CF Field: cell_area(latitude(5), longitude(8)) m2>
+        >>> print(a.array)
+        [[4.27128714e+12 4.27128714e+12 4.27128714e+12 4.27128714e+12
+          4.27128714e+12 4.27128714e+12 4.27128714e+12 4.27128714e+12]
+         [1.16693735e+13 1.16693735e+13 1.16693735e+13 1.16693735e+13
+          1.16693735e+13 1.16693735e+13 1.16693735e+13 1.16693735e+13]
+         [3.18813213e+13 3.18813213e+13 3.18813213e+13 3.18813213e+13
+          3.18813213e+13 3.18813213e+13 3.18813213e+13 3.18813213e+13]
+         [1.16693735e+13 1.16693735e+13 1.16693735e+13 1.16693735e+13
+          1.16693735e+13 1.16693735e+13 1.16693735e+13 1.16693735e+13]
+         [4.27128714e+12 4.27128714e+12 4.27128714e+12 4.27128714e+12
+          4.27128714e+12 4.27128714e+12 4.27128714e+12 4.27128714e+12]]
+        >>> f.cell_area(methods=True)
+        {(1,): 'linear longitude', (0,): 'linear sine latitude'}
+
         >>> a = f.cell_area(radius=cf.Data(3389.5, 'km'))
-        >>> a = f.cell_area(insert=True)
+
+        >>> c = f.cell_area(return_cell_measure=True)
+        >>> c
+        <CF CellMeasure: measure:area(5, 8) m2>
+        >>> f.set_construct(c)
+        'cellmeasure0'
+        >>> print(f)
+        Field: specific_humidity (ncvar%q)
+        ----------------------------------
+        Data            : specific_humidity(latitude(5), longitude(8)) 1
+        Cell methods    : area: mean
+        Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
+                        : longitude(8) = [22.5, ..., 337.5] degrees_east
+                        : time(1) = [2019-01-01 00:00:00]
+        Cell measures   : measure:area(latitude(5), longitude(8)) = [[4271287143027.272, ..., 4271287143027.272]] m2
+        >>> f.cell_area(methods=1)
```

Comment:
```suggestion
        >>> f.cell_area(methods=True)
```

1 is of course equivalent but could be confusing as an example.